c10/core/Scalar.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdexcept>
+#include <string>
+#include <utility>
+
+#include <c10/macros/Macros.h>
+#include <c10/core/ScalarType.h>
+#include <c10/Half.h>
+
+namespace at {
+class Tensor;
+}
+
+namespace c10 {
+
+class C10_API Scalar {
```

Comment:
Yes. Maybe something like:

Scalar represents a 0-dimensional tensor which contains a single element. Unlike a tensor, numeric literals (in C++) are implicitly convertible to Scalar (which is why, for example, we provide both `add(Tensor)` and `add(Scalar)` overloads for many operations). It may also be used in circumstances where you statically know a tensor is 0-dim and single size, but don't know it's type.